src/components/vanilla/MultiSelector.tsx

```diff
@@ -0,0 +1,247 @@
+
+export default (props: Props) => {
+  const ref = useRef<HTMLInputElement | null>(null);
+
+  const [focus, setFocus] = useState(false);
+  const [isDropdownOrItemFocused, setIsDropdownOrItemFocused] = useState(false);
+  const [search, setSearch] = useState('');
+  const [triggerBlur, setTriggerBlur] = useState(false);
+  const [value, setValue] = useState(props.defaultValue);
+
+  const [_, setServerSearch] = useEmbeddableState({
+    [props.searchProperty || 'search']: '',
+  }) as [SelectorRecord, (f: (m: SelectorRecord) => SelectorRecord) => void];
+
+  useEffect(() => {
+    setValue(props.defaultValue);
+  }, [props.defaultValue]);
+
+  // Accessibility - Close the menu if we've tabbed off of any items it contains
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout | null = null;
+
+    if (!isDropdownOrItemFocused) {
+      timeoutId = setTimeout(() => {
+        setFocus(false);
+      }, 200);
+    } else {
+      if (timeoutId) {
+        clearTimeout(timeoutId);
+      }
+    }
+
+    return () => {
+      if (timeoutId) {
+        clearTimeout(timeoutId);
+      }
+    };
+  }, [isDropdownOrItemFocused]);
+
+  useLayoutEffect(() => {
+    if (!triggerBlur) return;
+
+    const timeout = setTimeout(() => {
+      setFocus(false);
+      setTriggerBlur(false);
+    }, 500);
+
+    return () => clearTimeout(timeout);
+  }, [triggerBlur]);
+
+  const performSearch = useCallback(
+    (newSearch: string) => {
+      setSearch(newSearch);
+
+      clearTimeout(debounce);
+
+      debounce = window.setTimeout(() => {
+        setServerSearch((s) => ({ ...s, [props.searchProperty || 'search']: newSearch }));
+      }, 500);
+    },
+    [setSearch, setServerSearch, props.searchProperty],
+  );
+
+  const set = useCallback(
+    (newValue: string) => {
+      performSearch('');
+
+      let newValues: string[] = [];
+
+      if (newValue !== '') {
+        newValues = value || [];
+        if (newValues?.includes(newValue)) {
+          newValues = newValues.filter((v) => v !== newValue);
+        } else {
+          newValues = [...newValues, newValue];
+        }
+      }
+
+      props.onChange(newValues);
+      setValue(newValues);
+      setServerSearch((s) => ({ ...s, [props.searchProperty || 'search']: '' }));
+      clearTimeout(debounce);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [performSearch, props, value],
+  );
+
+  // Used for handling keydown events on the menu items
+  const handleKeyDownCallback = (
+    e: React.KeyboardEvent<HTMLElement>,
+    callback: any,
+    escapable?: boolean,
+  ) => {
+    if (e.key === 'Enter' || e.key === ' ') {
+      e.preventDefault();
+      callback(e);
+    }
+    if (escapable && e.key === 'Escape') {
+      e.preventDefault();
+      setFocus(false);
+      setTriggerBlur(true);
+    }
+  };
+
+  const list = useMemo(() => {
+    return props.options.map((option) => {
+      return (
+        <div
+          key={option.value}
```

Comment:
```suggestion
          key={option.value}
          role="button"
```
Can we give this a role button? Just for the sake of accessibility